runner/app/live/pipelines/segment_anything_2.py

```diff
@@ -0,0 +1,105 @@
+
+    model_id: str = "facebook/sam2-hiera-tiny"
+    point_coords: List[List[int]] = [[1, 1]]
+    point_labels: List[int] = [1]
+    show_points: Optional[bool] = True
+
+    def __init__(self, **data): 
+        super().__init__(**data)
+
+class Sam2Live(Pipeline):
+    def __init__(self, **params):
+        super().__init__(**params)
+        self.pipe: Optional[Sam2Wrapper] = None
+        self.first_frame = True
+        self.update_params(**params)
+
+    def update_params(self, **params):
+        new_params = Sam2LiveParams(**params)
+        self.params = new_params
+
+        logging.info(f"Setting parameters for sam2")
+        self.pipe = Sam2Wrapper(
+            model_id_or_path=self.params.model_id,
+            point_coords=self.params.point_coords,
+            point_labels=self.params.point_labels,
+            show_points=self.params.show_points,
+            
+            # Add additional as needed
+        )
+
+        self.params = new_params
+        self.first_frame = True
+
+    def _process_mask(self, mask: np.ndarray, frame_shape: tuple) -> np.ndarray:
+        """Process and resize mask if needed."""
+        if mask.shape[0] == 0:
+            return np.zeros((frame_shape[0], frame_shape[1]), dtype="uint8")
+            
+        mask = (mask[0, 0] > 0).cpu().numpy().astype("uint8") * 255
+        if mask.shape[:2] != frame_shape[:2]:
+            mask = cv2.resize(mask, (frame_shape[1], frame_shape[0]))
+        return mask
+
+    def process_frame(self, frame: Image.Image, **params) -> Image.Image:
+        frame_lock = threading.Lock()
+        try:
+            if params:
+                self.update_params(**params)
+
+            # Convert image formats
+            frame_array = np.array(frame)
+            frame_bgr = cv2.cvtColor(frame_array, cv2.COLOR_RGBA2BGR)
+            if self.first_frame:
+                self.pipe.predictor.load_first_frame(frame)
+                
+                for idx, point in enumerate(self.params.point_coords):
+                    _, _, mask_logits = self.pipe.predictor.add_new_prompt(
+                        frame_idx=0, 
+                        obj_id=idx + 1, 
+                        points=[point], 
+                        labels=[self.params.point_labels[idx]]
+                    )
+                self.first_frame = False
+            else:
+                _, mask_logits = self.pipe.predictor.track(frame)
+                
+            # Process mask and create overlay
+            mask = self._process_mask(mask_logits, frame_bgr.shape)
+
+            # Create an overlay by combining the original frame and the mask
+            colored_mask = cv2.cvtColor(mask, cv2.COLOR_GRAY2BGR)
+            colored_mask[mask > 0] = [255, 0, 255]  # Add purple tint to the mask
+            overlay = cv2.addWeighted(frame_bgr, 1, colored_mask, 1, 0)
+            # Draw points on the overlay
+            if hasattr(self.params, 'show_points') and self.params.show_points:
```

Comment:
`hasattr` is apparently a noop for Pydantic models, so it will always return `True` even if the field was not set. Pydantic initializes it with `None`, so we can just remove it here
```suggestion
            if self.params.show_points:
```